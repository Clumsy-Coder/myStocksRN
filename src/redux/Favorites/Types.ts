import { Action } from 'redux';

/*
  ██████╗ ███████╗██████╗ ██╗   ██╗ ██████╗███████╗██████╗     ███████╗████████╗ █████╗ ████████╗███████╗
  ██╔══██╗██╔════╝██╔══██╗██║   ██║██╔════╝██╔════╝██╔══██╗    ██╔════╝╚══██╔══╝██╔══██╗╚══██╔══╝██╔════╝
  ██████╔╝█████╗  ██║  ██║██║   ██║██║     █████╗  ██████╔╝    ███████╗   ██║   ███████║   ██║   █████╗
  ██╔══██╗██╔══╝  ██║  ██║██║   ██║██║     ██╔══╝  ██╔══██╗    ╚════██║   ██║   ██╔══██║   ██║   ██╔══╝
  ██║  ██║███████╗██████╔╝╚██████╔╝╚██████╗███████╗██║  ██║    ███████║   ██║   ██║  ██║   ██║   ███████╗
  ╚═╝  ╚═╝╚══════╝╚═════╝  ╚═════╝  ╚═════╝╚══════╝╚═╝  ╚═╝    ╚══════╝   ╚═╝   ╚═╝  ╚═╝   ╚═╝   ╚══════╝

*/

/**
 * Favorites reducer state
 */
export interface FavoritesReducerState {
  symbols: string[];
}

// ---------------------------------------------------------------------------------------------- //
/*
   █████╗  ██████╗████████╗██╗ ██████╗ ███╗   ██╗    ████████╗██╗   ██╗██████╗ ███████╗███████╗
  ██╔══██╗██╔════╝╚══██╔══╝██║██╔═══██╗████╗  ██║    ╚══██╔══╝╚██╗ ██╔╝██╔══██╗██╔════╝██╔════╝
  ███████║██║        ██║   ██║██║   ██║██╔██╗ ██║       ██║    ╚████╔╝ ██████╔╝█████╗  ███████╗
  ██╔══██║██║        ██║   ██║██║   ██║██║╚██╗██║       ██║     ╚██╔╝  ██╔═══╝ ██╔══╝  ╚════██║
  ██║  ██║╚██████╗   ██║   ██║╚██████╔╝██║ ╚████║       ██║      ██║   ██║     ███████╗███████║
  ╚═╝  ╚═╝ ╚═════╝   ╚═╝   ╚═╝ ╚═════╝ ╚═╝  ╚═══╝       ╚═╝      ╚═╝   ╚═╝     ╚══════╝╚══════╝

*/

/**
 * Favorites action types
 */
export enum ActionTypes {
  ADD_FAVORITE_STOCK = 'FAVORITES/ADD_FAVORITE_STOCK',
  REMOVE_FAVORITE_STOCK = 'FAVORITES/REMOVE_FAVORITE_STOCK',
  CLEAR_FAVORITE_STOCKS = 'FAVORITES/CLEAR_FAVORITE_STOCKS',
}

// ---------------------------------------------------------------------------------------------- //
/*
   █████╗  ██████╗████████╗██╗ ██████╗ ███╗   ██╗███████╗
  ██╔══██╗██╔════╝╚══██╔══╝██║██╔═══██╗████╗  ██║██╔════╝
  ███████║██║        ██║   ██║██║   ██║██╔██╗ ██║███████╗
  ██╔══██║██║        ██║   ██║██║   ██║██║╚██╗██║╚════██║
  ██║  ██║╚██████╗   ██║   ██║╚██████╔╝██║ ╚████║███████║
  ╚═╝  ╚═╝ ╚═════╝   ╚═╝   ╚═╝ ╚═════╝ ╚═╝  ╚═══╝╚══════╝

*/

/**
 * Add stock to Favorites action
 */
export interface AddFavoriteStockAction extends Action<ActionTypes.ADD_FAVORITE_STOCK> {
  type: ActionTypes.ADD_FAVORITE_STOCK;
  readonly stockSymbol: string;
}

/**
 * Remove stock from Favorites action
 */
export interface RemoveFavoriteStockAction extends Action<ActionTypes.REMOVE_FAVORITE_STOCK> {
  type: ActionTypes.REMOVE_FAVORITE_STOCK;
  readonly stockSymbol: string;
}

/**
 * Clear all stocks from Favorites action
 */
export interface ClearFavoriteStocksAction extends Action<ActionTypes.CLEAR_FAVORITE_STOCKS> {
  type: ActionTypes.CLEAR_FAVORITE_STOCKS;
}

/**
 * Union of actions for AddFavoriteStock, RemoveFavoriteStock, ClearFavoriteStocks
 */
export type FavoritesActions =
  | AddFavoriteStockAction
  | RemoveFavoriteStockAction
  | ClearFavoriteStocksAction;
