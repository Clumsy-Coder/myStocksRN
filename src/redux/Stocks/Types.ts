/**
 * Data format when fetching stock quote
 * check https://iexcloud.io/docs/api/#quote
 *
 * @interface IStockQuote
 */
export interface IStockQuote {
  symbol: string; // stock ticker. ex: AAPL
  companyName: string; // company name
  primaryExchange: string; // Refers to the primary listing exchange for the symbol
  latestPrice: number; // Refers to the latest relevant price of the security which is derived from multiple sources
  latestVolume: number; // Refers to the latest total market volume of the stock across all markets
  latestUpdate: number; // latest update time in epoch represented in milliseconds
  latestTime: string; // latest update time in human readable format
  change: number; // Refers to the change in price between latestPrice and previousClose
  changePercent: number; // Refers to the percent change in price between latestPrice and previousClose
  open: number; // open stock price
  close: number; // close stock price
  high: number | null; // highest stock price during stock hours. null before 9:45 and weekends
  low: number | null; // lowest stock price during stock hours. null before 9:45 and weekends
  week52High: number; // highest stock price in the last 52 weeks
  week52Low: number; // lowest stock price in the last 52 weeks
  marketCap: number; // is calculated in real time using latestPrice
  ytdChange: number; // Refers to the price change percentage from start of year to previous close
  peRatio: number; // Refers to the price-to-earnings ratio for the company
  // volume: number; // Total volume for the stock, but only updated after market open
  avgTotalVolume: number;
  previousClose: number; // Refers to the previous trading day closing price
  extendedPrice: number; // Refers to the 15 minute delayed price outside normal market hours 0400 - 0930 ET and 1600 - 2000 ET
  extendedChange: number; // Refers to the price change between extendedPrice and latestPrice
  extendedChangePercent: number; // Refers to the price change percent between extendedPrice and latestPrice
  previousVolume: number; // Refers to the previous trading day volume

  // openTime: number;
  // closeTime: number;
  // openSource: string;
  // closeSource: string;
  // highTime: number;
  // highSource: string;
  // lowTime: number;
  // lowSource: string;

  // delayedPrice: number;
  // delayedPriceTime: number;
  // oddLotDelayedPrice: number;
  // oddLotDelayedPriceTime: number;
  // extendedPriceTime: number;
  // iexRealtimePrice: number;
  // iexRealtimeSize: number;
  // iexLastUpdated: number;
  // iexMarketPercent: number;
  // iexVolume: number;
  // iexBidPrice: number;
  // iexBidSize: number;
  // iexAskPrice: number;
  // iexAskSize: number;
  // iexOpen: null;
  // iexOpenTime: null;
  // iexClose: number;
  // iexCloseTime: number;
  // lastTradeTime: number;
  // isUSMarketOpen: Boolean;
}

/**
 * Data format when fetching stock chart data.
 * check https://iexcloud.io/docs/api/#charts
 * check https://iexcloud.io/docs/api/#historical-prices
 * check https://iexcloud.io/docs/api/#intraday-prices
 *
 * @interface IStockChart
 */
export interface IStockChart {
  date: string; // Formatted as YYYY-MM-DD
  open: number; // Adjusted data for historical dates. Split adjusted only
  close: number; // Adjusted data for historical dates. Split adjusted only
  high: number; // Adjusted data for historical dates. Split adjusted only
  low: number; // Adjusted data for historical dates. Split adjusted only
  volume: number; // Adjusted data for historical dates. Split adjusted only
  currency: string;
  change: number; // Change from previous trading day
  changePercent: number; // Change percent from previous trading day
  label: string; // A human readable format of the date depending on the range
  changeOverTime: number; // Percent change of each interval relative to first value. Useful for comparing multiple stocks
  // uClose: number; // Unadjusted data for historical dates
  // uOpen: number; // Unadjusted data for historical dates
  // uHigh: number; // Unadjusted data for historical dates
  // uLow: number; // Unadjusted data for historical dates
  // uVolume: number; // Unadjusted data for historical dates
}

// ---------------------------------------------------------------------------------------------- //
// used for Stocks reducer
export interface IStockState {
  [symbol: string]: {
    quote: {
      fetching: Boolean;
      data: IStockQuote | null;
      error?: any;
    };
    chart?: {
      fetching: boolean;
      data: IStockChart[] | null;
      error?: any;
    };
  };
}

// ---------------------------------------------------------------------------------------------- //
// action types

export enum ActionTypesEnum {
  GET_STOCK_QUOTE = 'STOCKS/GET_STOCK_QUOTE',
  GET_STOCK_QUOTE_PENDING = 'STOCKS/GET_STOCK_QUOTE_PENDING',
  GET_STOCK_QUOTE_FULFILLED = 'STOCKS/GET_STOCK_QUOTE_FULFILLED',
  GET_STOCK_QUOTE_REJECTED = 'STOCKS/GET_STOCK_QUOTE_REJECTED',

  GET_STOCK_CHART = 'STOCKS/GET_STOCK_CHART',
  GET_STOCK_CHART_PENDING = 'STOCKS/GET_STOCK_CHART_PENDING',
  GET_STOCK_CHART_FULFILLED = 'STOCKS/GET_STOCK_CHART_FULFILLED',
  GET_STOCK_CHART_REJECTED = 'STOCKS/GET_STOCK_CHART_REJECTED',
}

// ---------------------------------------------------------------------------------------------- //
// action creators

// get stock quote action creator
// called to initiate the fetching process
export interface IGetStockQuoteAC {
  type: ActionTypesEnum.GET_STOCK_QUOTE;
  readonly stockSymbol: string; // stock symbol
}

// called only in redux saga to handle side effects
export interface IGetStockQuotePendingAC {
  type: ActionTypesEnum.GET_STOCK_QUOTE_PENDING;
  readonly stockSymbol: string;
}

// called only in redux saga to handle side effects
export interface IGetStockQuoteFulfilledAC {
  type: ActionTypesEnum.GET_STOCK_QUOTE_FULFILLED;
  readonly stockSymbol: string;
  readonly payload: { data: IStockQuote };
}

// called only in redux saga to handle side effects
export interface IGetStockQuoteRejectedAC {
  type: ActionTypesEnum.GET_STOCK_QUOTE_REJECTED;
  readonly stockSymbol: string;
  readonly error: any;
}

export type StockQuoteAction =
  | IGetStockQuoteAC
  | IGetStockQuotePendingAC
  | IGetStockQuoteFulfilledAC
  | IGetStockQuoteRejectedAC;

// ---------------------------------------------------------------------------------------------- //
// get stock chart action creator
// called to initiate the fetching process
export interface IGetStockChartAC {
  type: ActionTypesEnum.GET_STOCK_CHART;
  readonly stockSymbol: string;
}

// called only in redux saga to handle side effects
export interface IGetStockChartPendingAC {
  type: ActionTypesEnum.GET_STOCK_CHART_PENDING;
  readonly stockSymbol: string;
}

// called only in redux saga to handle side effects
export interface IGetStockChartFulfilledAC {
  type: ActionTypesEnum.GET_STOCK_CHART_FULFILLED;
  readonly stockSymbol: string;
  readonly payload: { data: IStockChart[] };
}

// called only in redux saga to handle side effects
export interface IGetStockChartRejectedAC {
  type: ActionTypesEnum.GET_STOCK_CHART_REJECTED;
  readonly stockSymbol: string;
  readonly error: any;
}

export type StockChartAction =
  | IGetStockChartAC
  | IGetStockChartPendingAC
  | IGetStockChartFulfilledAC
  | IGetStockChartRejectedAC;

// ---------------------------------------------------------------------------------------------- //
// all Stock action creators
export type StockActionTypes = StockQuoteAction | StockChartAction;
