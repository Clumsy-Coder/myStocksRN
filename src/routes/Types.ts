import { StackNavigationProp } from '@react-navigation/stack';
import { RouteProp } from '@react-navigation/native';

/**
 * All navigation routes available
 */
export enum NavigationRoutePath {
  Home = 'Home',
  StockDetails = 'StockDetails',
  About = 'About',
}

/**
 * Parameters for each Screen
 */
export type RootStackParamList = {
  [NavigationRoutePath.Home]: undefined;
  [NavigationRoutePath.StockDetails]: {
    symbol: string;
  };
  [NavigationRoutePath.About]: undefined;
};

/**
 * StockDetails **navigation** prop.
 * Used when navigating to StockDetails screen
 */
export type StockDetailsNavigationProp = StackNavigationProp<
  RootStackParamList,
  NavigationRoutePath.StockDetails
>;

/**
 * StockDetails **route** prop.
 * Used in StockDetails screen.
 * Primarily used for getting the param value
 */
export type StockDetailsRouteProp = RouteProp<RootStackParamList, NavigationRoutePath.StockDetails>;
