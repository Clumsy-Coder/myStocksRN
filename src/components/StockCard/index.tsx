import React from 'react';
import { View, StyleSheet } from 'react-native';
import { ListItem, Text, H1, H3 } from 'native-base';
import { useNavigation } from '@react-navigation/native';

import { Selectors } from 'src/redux/Stocks/Types';
import { StockDetailsNavigationProp, NavigationRoutePath } from '@routes/Types';

interface OwnProps {
  data: Selectors.SelectQuoteTrim;
}

const styleSheet = StyleSheet.create({
  // main container
  container: {
    // backgroundColor: 'blue',
  },
  listItemContainer: {
    flex: 1,
    flexDirection: 'row',
    height: 70,
  },
  // company
  companyContainer: {
    flex: 3,
    flexDirection: 'column',
    // backgroundColor: 'orange',
    justifyContent: 'space-around',
  },
  companyStockSymbol: {
    fontWeight: 'bold',
  },
  companySymbol: {
    fontWeight: 'bold',
  },
  companyName: {},
  // price
  priceContainer: {
    flex: 1,
    flexDirection: 'column',
    // backgroundColor: 'pink',
    justifyContent: 'space-around',
    alignItems: 'flex-end',
  },
  priceValue: {},
  priceChangeContainer: {
    borderRadius: 4,
    padding: 1,
    paddingVertical: 2,
    paddingRight: 5,
    width: '100%',
    alignItems: 'flex-end',
  },
  priceChange: {
    color: 'white',
  },
  priceRed: {
    backgroundColor: 'red',
  },
  priceGreen: {
    backgroundColor: 'green',
  },
});

const StockCard: React.FC<OwnProps> = (props: OwnProps) => {
  const { data } = props;
  const navigation = useNavigation<StockDetailsNavigationProp>();

  if (data.fetching) {
    return (
      <ListItem>
        <Text>Loading</Text>
      </ListItem>
    );
  }

  const change = data.change === undefined ? '0' : data.change;

  return (
    <ListItem
      style={styleSheet.container}
      onPress={(): void =>
        navigation.navigate(NavigationRoutePath.StockDetails, { symbol: data.symbol })
      }
    >
      <View style={styleSheet.listItemContainer}>
        <View style={styleSheet.companyContainer}>
          <H1 style={styleSheet.companySymbol}>{data.symbol}</H1>
          <H3 style={styleSheet.companyName}>{data.companyName}</H3>
        </View>
        <View style={styleSheet.priceContainer}>
          <H3>{data.price.toFixed(2)}</H3>
          <View
            style={[
              styleSheet.priceChangeContainer,
              +change > 0 ? styleSheet.priceGreen : styleSheet.priceRed,
            ]}
          >
            <H3 style={styleSheet.priceChange}>{data.change}</H3>
          </View>
        </View>
      </View>
    </ListItem>
  );
};

export default StockCard;
