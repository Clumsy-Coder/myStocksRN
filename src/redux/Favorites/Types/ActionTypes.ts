/**
 * Favorites action types
 */
enum ActionTypes {
  ADD_FAVORITE_STOCK = 'FAVORITES/ADD_FAVORITE_STOCK',
  REMOVE_FAVORITE_STOCK = 'FAVORITES/REMOVE_FAVORITE_STOCK',
  CLEAR_FAVORITE_STOCKS = 'FAVORITES/CLEAR_FAVORITE_STOCKS',
}

export default ActionTypes;
