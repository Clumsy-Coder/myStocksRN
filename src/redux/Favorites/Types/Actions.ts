import { Action } from 'redux';

import ActionTypes from '@redux/Favorites/Types/ActionTypes';

/**
 * Add stock to Favorites action
 */
export interface AddFavoriteStockAction extends Action<ActionTypes.ADD_FAVORITE_STOCK> {
  type: ActionTypes.ADD_FAVORITE_STOCK;
  readonly stockSymbol: string;
}

/**
 * Remove stock from Favorites action
 */
export interface RemoveFavoriteStockAction extends Action<ActionTypes.REMOVE_FAVORITE_STOCK> {
  type: ActionTypes.REMOVE_FAVORITE_STOCK;
  readonly stockSymbol: string;
}

/**
 * Clear all stocks from Favorites action
 */
export interface ClearFavoriteStocksAction extends Action<ActionTypes.CLEAR_FAVORITE_STOCKS> {
  type: ActionTypes.CLEAR_FAVORITE_STOCKS;
}

/**
 * Union of actions for AddFavoriteStock, RemoveFavoriteStock, ClearFavoriteStocks
 */
export type FavoritesActions =
  | AddFavoriteStockAction
  | RemoveFavoriteStockAction
  | ClearFavoriteStocksAction;
