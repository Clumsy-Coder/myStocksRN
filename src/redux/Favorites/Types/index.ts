import ActionTypes from '@redux/Favorites/Types/ActionTypes';
import * as Actions from '@redux/Favorites/Types/Actions';
import * as Reducer from '@redux/Favorites/Types/Reducer';

export { ActionTypes };
export { Actions };
export { Reducer };
