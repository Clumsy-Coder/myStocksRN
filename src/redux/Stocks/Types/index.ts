import ActionTypes from '@redux/Stocks/Types/ActionTypes';
import * as Actions from '@redux/Stocks/Types/Actions';
import * as DataDomain from '@redux/Stocks/Types/DataDomain';
import * as Selectors from '@redux/Stocks/Types/Selectors';
import * as Reducer from '@redux/Stocks/Types/Reducer';

export { ActionTypes };
export { Actions };
export { DataDomain };
export { Selectors };
export { Reducer };
