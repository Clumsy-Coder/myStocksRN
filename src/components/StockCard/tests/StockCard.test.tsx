import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';

import StockCard from 'src/components/StockCard';
import { Selectors } from 'src/redux/Stocks/Types';

const stockSymbol1 = 'IBM';
const companyName1 = 'International Business Machines Corporation';
const stockCardProps1: Selectors.SelectQuoteTrim = {
  fetching: false,
  symbol: stockSymbol1,
  companyName: companyName1,
  price: '100',
  change: '1.0',
  changePercent: '10.0',
};

const stockCardProps2: Selectors.SelectQuoteTrim = {
  fetching: true,
  symbol: stockSymbol1,
  companyName: companyName1,
  price: undefined,
  change: undefined,
  changePercent: undefined,
};

const stockCardProps3: Selectors.SelectQuoteTrim = {
  fetching: false,
  symbol: stockSymbol1,
  companyName: companyName1,
  price: '100',
  change: undefined,
  changePercent: '10.0',
};

const stockCardProps4: Selectors.SelectQuoteTrim = {
  fetching: false,
  symbol: stockSymbol1,
  companyName: companyName1,
  price: '100',
  change: '-1.0',
  changePercent: '-10.0',
};

const stockCardProps5: Selectors.SelectQuoteTrim = {
  fetching: false,
  symbol: stockSymbol1,
  companyName: companyName1,
  price: '100',
  change: '1.0',
  changePercent: '10.0',
};

describe('<StockCard />', () => {
  describe('During fetching data', () => {
    const wrapper = shallow(<StockCard data={stockCardProps2} />);

    it('Renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('After fetching data', () => {
    describe('Full data presented', () => {
      const wrapper = shallow(<StockCard data={stockCardProps1} />);

      it('Renders as expected', () => {
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('Undefined stock price', () => {
      const wrapper = shallow(<StockCard data={stockCardProps3} />);

      it('Renders as expected', () => {
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('Stock price less than 0', () => {
      const wrapper = shallow(<StockCard data={stockCardProps4} />);

      it('Renders as expected', () => {
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('Stock price greater than 0', () => {
      const wrapper = shallow(<StockCard data={stockCardProps5} />);

      it('Renders as expected', () => {
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
