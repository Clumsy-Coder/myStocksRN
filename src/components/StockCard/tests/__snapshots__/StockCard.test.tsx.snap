// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StockCard /> After fetching data Full data presented Renders as expected 1`] = `
<Styled(ListItem)
  style={Object {}}
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "height": 70,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 3,
          "flexDirection": "column",
          "justifyContent": "space-around",
        }
      }
    >
      <Styled(H1)
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        IBM
      </Styled(H1)>
      <Styled(H3)
        style={Object {}}
      >
        International Business Machines Corporation
      </Styled(H3)>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-around",
        }
      }
    >
      <Styled(H3)>
        100
      </Styled(H3)>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "borderRadius": 4,
              "padding": 1,
              "paddingRight": 5,
              "paddingVertical": 2,
              "width": "100%",
            },
            Object {
              "backgroundColor": "green",
            },
          ]
        }
      >
        <Styled(H3)
          style={
            Object {
              "color": "white",
            }
          }
        >
          1.0
        </Styled(H3)>
      </View>
    </View>
  </View>
</Styled(ListItem)>
`;

exports[`<StockCard /> After fetching data Stock price greater than 0 Renders as expected 1`] = `
<Styled(ListItem)
  style={Object {}}
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "height": 70,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 3,
          "flexDirection": "column",
          "justifyContent": "space-around",
        }
      }
    >
      <Styled(H1)
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        IBM
      </Styled(H1)>
      <Styled(H3)
        style={Object {}}
      >
        International Business Machines Corporation
      </Styled(H3)>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-around",
        }
      }
    >
      <Styled(H3)>
        100
      </Styled(H3)>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "borderRadius": 4,
              "padding": 1,
              "paddingRight": 5,
              "paddingVertical": 2,
              "width": "100%",
            },
            Object {
              "backgroundColor": "green",
            },
          ]
        }
      >
        <Styled(H3)
          style={
            Object {
              "color": "white",
            }
          }
        >
          1.0
        </Styled(H3)>
      </View>
    </View>
  </View>
</Styled(ListItem)>
`;

exports[`<StockCard /> After fetching data Stock price less than 0 Renders as expected 1`] = `
<Styled(ListItem)
  style={Object {}}
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "height": 70,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 3,
          "flexDirection": "column",
          "justifyContent": "space-around",
        }
      }
    >
      <Styled(H1)
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        IBM
      </Styled(H1)>
      <Styled(H3)
        style={Object {}}
      >
        International Business Machines Corporation
      </Styled(H3)>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-around",
        }
      }
    >
      <Styled(H3)>
        100
      </Styled(H3)>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "borderRadius": 4,
              "padding": 1,
              "paddingRight": 5,
              "paddingVertical": 2,
              "width": "100%",
            },
            Object {
              "backgroundColor": "red",
            },
          ]
        }
      >
        <Styled(H3)
          style={
            Object {
              "color": "white",
            }
          }
        >
          -1.0
        </Styled(H3)>
      </View>
    </View>
  </View>
</Styled(ListItem)>
`;

exports[`<StockCard /> After fetching data Undefined stock price Renders as expected 1`] = `
<Styled(ListItem)
  style={Object {}}
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "height": 70,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 3,
          "flexDirection": "column",
          "justifyContent": "space-around",
        }
      }
    >
      <Styled(H1)
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        IBM
      </Styled(H1)>
      <Styled(H3)
        style={Object {}}
      >
        International Business Machines Corporation
      </Styled(H3)>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-around",
        }
      }
    >
      <Styled(H3)>
        100
      </Styled(H3)>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "borderRadius": 4,
              "padding": 1,
              "paddingRight": 5,
              "paddingVertical": 2,
              "width": "100%",
            },
            Object {
              "backgroundColor": "red",
            },
          ]
        }
      >
        <Styled(H3)
          style={
            Object {
              "color": "white",
            }
          }
        />
      </View>
    </View>
  </View>
</Styled(ListItem)>
`;

exports[`<StockCard /> During fetching data Renders as expected 1`] = `
<Styled(ListItem)>
  <Styled(Text)>
    Loading
  </Styled(Text)>
</Styled(ListItem)>
`;
